storages/merger/listV2.go

```diff
@@ -0,0 +1,125 @@
+package merger
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/allegro/akubra/log"
+	"github.com/allegro/akubra/storages/merger/s3datatypes"
+	"github.com/allegro/akubra/transport"
+)
+
+// MergeBucketListV2Responses unifies responses from multiple backends
+func MergeBucketListV2Responses(successes []transport.ResErrTuple) (resp *http.Response, err error) {
+	if len(successes) == 0 {
+		err = fmt.Errorf("No successful responses")
+		return
+	}
+	keys := objectsContainer{
+		list: make([]fmt.Stringer, 0),
+		set:  make(map[string]struct{}),
+	}
+	prefixes := objectsContainer{
+		list: make([]fmt.Stringer, 0),
+		set:  make(map[string]struct{}),
+	}
+	var listBucketV2Result s3datatypes.ListBucketV2Result
+	for _, tuple := range successes {
+		resp = tuple.Res
+		listBucketV2Result = extractListv2Results(resp)
+		keys.append(listBucketV2Result.Contents.ToStringer()...)
+		prefixes.append(listBucketV2Result.CommonPrefixes.ToStringer()...)
+	}
+
+	req := successes[0].Res.Request
+	reqQuery := req.URL.Query()
+	maxKeysQuery := reqQuery.Get("max-keys")
+	maxKeys, err := strconv.Atoi(maxKeysQuery)
+	if err != nil {
+		maxKeys = 1000
+	}
+
+	listBucketV2Result = pickListV2ResultSet(keys, prefixes, maxKeys, listBucketV2Result)
+
+	bodyBytes, err := xml.Marshal(listBucketV2Result)
+	if err != nil {
+		log.Debug("Problem marshalling ObjectStore response body, %s", err)
+		return nil, err
+	}
+	buf := bytes.NewBuffer(bodyBytes)
+	resp.Body = ioutil.NopCloser(buf)
```

Comment:
Shouldn't there be somewhere an initialization of the `resp`? Also, what about other headers that are already in `successes[...].Res`? Shouldn't we merge them somehow?

Reply:
`resp` is one of valid backend responses (see line 32), so headers are appropriate